### ISSUE TITLE
A blank screen when booting with Glacier UI

### ISSUE THREAD

**Issue by marmistrz** [nemomobile-ux/glacier-home#67]
```
$ sudo journalctl | grep lipstick
nemo's password:
Feb 18 19:32:06 Sailfish systemd[1689]: Starting The lipstick UI...
Feb 18 19:32:07 Sailfish lipstick[1767]: [D] unknown:0 - EGLFS: Screen Info
Feb 18 19:32:07 Sailfish lipstick[1767]: [D] unknown:0 -  - Physical size: QSizeF(68, 122)
Feb 18 19:32:07 Sailfish lipstick[1767]: [D] unknown:0 -  - Screen size: QSize(1080, 1920)
Feb 18 19:32:07 Sailfish lipstick[1767]: [D] unknown:0 -  - Screen depth: 32
Feb 18 19:32:07 Sailfish lipstick[1767]: library "libgui.so" wasn't loaded and RTLD_NOLOAD prevented it
Feb 18 19:32:08 Sailfish lipstick[1767]: == hwcomposer module ==
Feb 18 19:32:08 Sailfish lipstick[1767]: * Address: 0xe4ded004
Feb 18 19:32:08 Sailfish lipstick[1767]: * Module API Version: 2
Feb 18 19:32:08 Sailfish lipstick[1767]: * HAL API Version: 0
Feb 18 19:32:08 Sailfish lipstick[1767]: * Identifier: hwcomposer
Feb 18 19:32:08 Sailfish lipstick[1767]: * Name: QTI Hardware Composer Module
Feb 18 19:32:08 Sailfish lipstick[1767]: * Author: CodeAurora Forum
Feb 18 19:32:08 Sailfish lipstick[1767]: == hwcomposer module ==
Feb 18 19:32:08 Sailfish lipstick[1767]: cannot locate symbol "_ZN3sdm3Sys7access_E" referenced by "/system/vendor/lib/libsdmextension.so"...
Feb 18 19:32:08 Sailfish lipstick[1767]: cannot locate symbol "_ZN3sdm3Sys7access_E" referenced by "/system/vendor/lib/libsdmextension.so"...
Feb 18 19:32:08 Sailfish lipstick[1767]: library "libtinyxml2_1.so" not found
Feb 18 19:32:08 Sailfish lipstick[1767]: cannot locate symbol "_ZN3sdm3Sys7access_E" referenced by "/system/vendor/lib/libsdmextension.so"...
Feb 18 19:32:08 Sailfish lipstick[1767]: cannot locate symbol "_ZN3sdm3Sys7access_E" referenced by "/system/vendor/lib/libsdmextension.so"...
Feb 18 19:32:08 Sailfish lipstick[1767]: == hwcomposer device ==
Feb 18 19:32:08 Sailfish lipstick[1767]: * Version: 1050001 (interpreted as 1050001)
Feb 18 19:32:08 Sailfish lipstick[1767]: * Module: 0xe4ded004
Feb 18 19:32:08 Sailfish lipstick[1767]: == hwcomposer device ==
Feb 18 19:32:08 Sailfish lipstick[1767]: [W] unknown:0 - QEglScreen 0x773cc8
Feb 18 19:32:09 Sailfish lipstick[1767]: [D] unknown:0 - Hardware Compositor support is disabled
Feb 18 19:32:09 Sailfish lipstick[1767]: [W] unknown:47 - file:///usr/share/lipstick-glacier-home-qt5/qml/compositor/ScreenGestureArea.qml:47: TypeError: Cannot read property 'window' of null
Feb 18 19:32:09 Sailfish lipstick[1767]: [W] unknown:337 - file:///usr/share/lipstick-glacier-home-qt5/qml/compositor.qml:337: TypeError: Cannot read property 'lockscreen' of undefined
Feb 18 19:32:09 Sailfish lipstick[1767]: [D] unknown:0 - Height:  1920 Width:  1080
Feb 18 19:32:09 Sailfish lipstick[1767]: [D] unknown:0 - Scale ratio:  2.24824  Font:  2.2
Feb 18 19:32:09 Sailfish lipstick[1767]: [D] unknown:0 - DPI is  401.575
Feb 18 19:32:09 Sailfish lipstick[1767]: [W] unknown:0 - QT_QPA_EGLFS_PHYSICAL_HEIGHT is not set!
Feb 18 19:32:09 Sailfish lipstick[1767]: [W] unknown:0 - QT_QPA_EGLFS_PHYSICAL_WIDTH is not set!
Feb 18 19:32:09 Sailfish lipstick[1767]: [W] unknown:0 - Device mm sizing don`t work
Feb 18 19:32:09 Sailfish lipstick[1767]: [D] unknown:0 - DP scale factor is  1.5
Feb 18 19:32:09 Sailfish lipstick[1767]: [D] unknown:0 - Height:  1920 Width:  1080
Feb 18 19:32:09 Sailfish lipstick[1767]: [D] unknown:0 - Scale ratio:  2.24824  Font:  2.2
Feb 18 19:32:09 Sailfish lipstick[1767]: [D] unknown:0 - DPI is  401.575
Feb 18 19:32:09 Sailfish lipstick[1767]: [W] unknown:0 - QT_QPA_EGLFS_PHYSICAL_HEIGHT is not set!
Feb 18 19:32:09 Sailfish lipstick[1767]: [W] unknown:0 - QT_QPA_EGLFS_PHYSICAL_WIDTH is not set!
Feb 18 19:32:09 Sailfish lipstick[1767]: [W] unknown:0 - Device mm sizing don`t work
Feb 18 19:32:09 Sailfish lipstick[1767]: [D] unknown:0 - DP scale factor is  1.5
Feb 18 19:32:09 Sailfish lipstick[1767]: [D] unknown:0 - MLocalThemeDaemonClient::MLocalThemeDaemonClient(const QString&, QObject*) Theme:  glacier  (hardcoded)
Feb 18 19:32:09 Sailfish lipstick[1767]: [D] unknown:0 - MLocalThemeDaemonClient::MLocalThemeDaemonClient(const QString&, QObject*) "/usr/share/themes/glacier/index.theme"
Feb 18 19:32:09 Sailfish lipstick[1767]: [D] unknown:0 - MLocalThemeDaemonClient::MLocalThemeDaemonClient(const QString&, QObject*) "/usr/share/themes/darko/index.theme"
Feb 18 19:32:09 Sailfish lipstick[1767]: [W] unknown:0 - MLocalThemeDaemonClient::MLocalThemeDaemonClient(const QString&, QObject*) Theme "glacier"  is invalid
Feb 18 19:32:09 Sailfish lipstick[1767]: [D] unknown:0 - LocalThemeDaemonClient: Looking for assets in ("/usr/share/themes/glacier/fontawesome", "/usr/share/themes/glacier/meegotouch")
Feb 18 19:32:10 Sailfish lipstick[1767]: [W] unknown:173 - file:///usr/share/lipstick-glacier-home-qt5/qml/MainScreen.qml:173:9: Type DeviceLockUI unavailable 
Feb 18 19:32:10 Sailfish lipstick[1767]: [W] unknown:49 - file:///usr/share/lipstick-glacier-home-qt5/qml/DeviceLockUI.qml:49:9: Cannot assign to non-existent property "onFeedback" 
Feb 18 19:32:10 Sailfish lipstick[1767]: [D] onWindowAdded:401 - Compositor: Window added "Home" category: 
Feb 18 19:32:10 Sailfish lipstick[1767]: [W] unknown:42 - file:///usr/share/lipstick-glacier-home-qt5/qml/notifications/NotificationPreview.qml:42: TypeError: Cannot read property 'parent' of undefined
Feb 18 19:32:10 Sailfish lipstick[1767]: [W] unknown:41 - file:///usr/share/lipstick-glacier-home-qt5/qml/notifications/NotificationPreview.qml:41: TypeError: Cannot read property 'parent' of undefined
Feb 18 19:32:10 Sailfish lipstick[1767]: [W] unknown:40 - file:///usr/share/lipstick-glacier-home-qt5/qml/notifications/NotificationPreview.qml:40: TypeError: Cannot read property 'parent' of undefined
Feb 18 19:32:10 Sailfish lipstick[1767]: [W] unknown:39 - file:///usr/share/lipstick-glacier-home-qt5/qml/notifications/NotificationPreview.qml:39: TypeError: Cannot read property 'parent' of undefined
Feb 18 19:32:10 Sailfish lipstick[1767]: [W] unknown:38 - file:///usr/share/lipstick-glacier-home-qt5/qml/notifications/NotificationPreview.qml:38: TypeError: Cannot read property 'parent' of undefined
Feb 18 19:32:10 Sailfish lipstick[1767]: [W] unknown:40 - file:///usr/share/lipstick-glacier-home-qt5/qml/volumecontrol/VolumeControl.qml:40: TypeError: Cannot read property of null
Feb 18 19:32:10 Sailfish lipstick[1767]: [W] unknown:0 - QML file not found: "connectivity/VpnAgent.qml"
Feb 18 19:32:10 Sailfish lipstick[1767]: [W] unknown:0 - QML search path: ("/usr/share/lipstick-glacier-home-qt5/qml")
Feb 18 19:32:10 Sailfish lipstick[1767]: [W] unknown:-1 - <Unknown File>: Invalid empty URL
Feb 18 19:32:10 Sailfish lipstick[1767]: [W] unknown:0 - QEglWindow 0x887c48: 0x807f38 0x0
Feb 18 19:32:10 Sailfish systemd[1689]: Started The lipstick UI.
Feb 18 19:32:32 Sailfish lipstick[1767]: [W] unknown:396 - file:///usr/share/lipstick-glacier-home-qt5/qml/compositor.qml:396: TypeError: Type error
Feb 18 19:32:32 Sailfish lipstick[1767]: [D] unknown:0 - sleepDisplay
Feb 18 19:33:38 Sailfish lipstick[1767]: [D] unknown:0 - unsleepDisplay
Feb 18 19:33:39 Sailfish lipstick[1767]: [W] unknown:396 - file:///usr/share/lipstick-glacier-home-qt5/qml/compositor.qml:396: TypeError: Type error
Feb 18 19:33:40 Sailfish lipstick[1767]: [D] unknown:0 - sleepDisplay
Feb 18 19:33:53 Sailfish lipstick[1767]: [D] onWindowAdded:401 - Compositor: Window added "Notification" category: notification
Feb 18 19:33:53 Sailfish lipstick[1767]: [D] unknown:0 - Unknown theme image: "icon-system-usb"
Feb 18 19:33:53 Sailfish lipstick[1767]: [D] unknown:0 - Unknown theme image: "icon-system-usb"
Feb 18 19:33:53 Sailfish lipstick[1767]: [W] unknown:123 - file:///usr/share/lipstick-glacier-home-qt5/qml/notifications/NotificationPreview.qml:123:13: QML Image: Failed to get image from provider: image://theme/icon-system-usb
Feb 18 19:33:53 Sailfish mce[1518]: tklock.c: tklock_dbus_notification_beg_cb(): notification begin from name=:1.29 owner=:1.29 pid=1767 uid=100000 gid=100000 priv=0 cmd=/usr/bin/lipstick -plugin evdevtouch:/dev/input/event4 -plugin
Feb 18 19:33:53 Sailfish lipstick[1767]: [D] unknown:0 - unsleepDisplay
Feb 18 19:33:57 Sailfish mce[1518]: tklock.c: tklock_dbus_notification_beg_cb(): notification begin from name=:1.29 owner=:1.29 pid=1767 uid=100000 gid=100000 priv=0 cmd=/usr/bin/lipstick -plugin evdevtouch:/dev/input/event4 -plugin
Feb 18 19:34:00 Sailfish lipstick[1767]: [D] onWindowRemoved:463 - Compositor: Window removed "Notification" category: notification
Feb 18 19:34:00 Sailfish lipstick[1767]: [W] unknown:464 - file:///usr/share/lipstick-glacier-home-qt5/qml/compositor.qml:464: TypeError: Cannot read property 'switcher' of undefined
Feb 18 19:34:00 Sailfish lipstick[1767]: [W] unknown:29 - file:///usr/share/lipstick-glacier-home-qt5/qml/compositor/WindowWrapperBase.qml:29: TypeError: Cannot read property of null
Feb 18 19:34:00 Sailfish lipstick[1767]: [W] unknown:30 - file:///usr/share/lipstick-glacier-home-qt5/qml/compositor/WindowWrapperBase.qml:30: TypeError: Cannot read property of null
Feb 18 19:34:03 Sailfish lipstick[1767]: [W] unknown:396 - file:///usr/share/lipstick-glacier-home-qt5/qml/compositor.qml:396: TypeError: Type error
Feb 18 19:34:03 Sailfish lipstick[1767]: [D] unknown:0 - sleepDisplay
```

SailfishOS 2.1.3.7

**Comment by p1pahare** [comment]
intex aqua fish with Sfos v 2.1.3.7

**Comment by marmistrz** [comment]
OnePlus 3, delaya73's port.
Other device users have reported the same in the TMO thread

**Comment by neochapay** [comment]
@marmistrz @p1pahare with repo you use ?

**Comment by marmistrz** [comment]
nemo-devel-ux

**Comment by eekkelund** [comment]
If I am correct this is because [SFOS 2.1.1](https://together.jolla.com/question/164350/changelog-211-jamsanjoki/#164350-jolla-devicelock) (?) devicelock change.
@M2ko has been working on the fix I assume, I can ask it from him.

**Comment by neochapay** [comment]
@eekkelund i think problem here:
`Feb 18 19:32:10 Sailfish lipstick[1767]: [W] unknown:0 - QML search path: ("/usr/share/lipstick-glacier-home-qt5/qml")
Feb 18 19:32:10 Sailfish lipstick[1767]: [W] unknown:-1 - <Unknown File>: Invalid empty URL
Feb 18 19:32:10 Sailfish lipstick[1767]: [W] unknown:0 - QEglWindow 0x887c48: 0x807f38 0x0`

**Comment by eekkelund** [comment]
My guess is 
`W] unknown:173 - file:///usr/share/lipstick-glacier-home-qt5/qml/MainScreen.qml:173:9: Type DeviceLockUI unavailable 
Feb 18 19:32:10 Sailfish lipstick[1767]: [W] unknown:49 - file:///usr/share/lipstick-glacier-home-qt5/qml/DeviceLockUI.qml:49:9: Cannot assign to non-existent property "onFeedback" `

I will check/test @M2ko's fix and report back!

**Comment by neochapay** [comment]
Fix in #68 but need check

**Comment by marmistrz** [comment]
I should have some platform sdk installed, I'll build the rpm and check.
